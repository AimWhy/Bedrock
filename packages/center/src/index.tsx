import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
import PropTypes from "prop-types";
import styled from "styled-components";

export interface CenterProps {
  maxWidth?: number | string;
  centerText?: boolean;
  centerChildren?: boolean;
}

function getSafeMaxWidth(
  breakPoints: Record<string, unknown>,
  maxWidth?: number | string
) {
  if (typeof maxWidth === "string") {
    return maxWidth;
  }

  return typeof maxWidth === "number"
    ? `${maxWidth}px`
    : mergeBreakpoints(breakPoints).medium + "px";
}

export const Center = styled.div.attrs<CenterProps>(() => {
  return {
    "data-bedrock-layout-center": "",
  };
})<CenterProps>`
  @property --maxWidth {
    syntax: "<length>";
    inherits: false;
    initial-value: 1023px;
  }
  --maxWidth: ${({ maxWidth, theme: { breakPoints } }) =>
    getSafeMaxWidth(breakPoints, maxWidth)};

  box-sizing: content-box;

  margin-inline-start: auto;
  margin-inline-end: auto;
  margin-inline: auto;

  max-inline-size: 1023px;

  @supports (
    max-inline-size:
      ${({ maxWidth, theme: { breakPoints } }) =>
        getSafeMaxWidth(breakPoints, maxWidth)}
  ) {
    max-inline-size: var(--maxWidth, 1023px);
  }

  ${(props) => (props.centerText ? "text-align: center;" : "")}

  ${(props) =>
    props.centerChildren
      ? `
  display: flex;
  flex-direction: column;
  align-items: center;
  `
      : ""}
`;

Center.displayName = "Center";

Center.propTypes = {
  maxWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  centerText: PropTypes.bool,
};
