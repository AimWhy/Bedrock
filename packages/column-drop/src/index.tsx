import {
  CSSLength,
  Gutter,
  SizesOptions,
  getSafeGutter,
  getSizeValue,
  useTheme,
} from "@bedrock-layout/spacing-constants";
import { forwardRefWithAs } from "@bedrock-layout/type-utils";
import React, { CSSProperties } from "react";

type MinItemWidth = CSSLength | number | SizesOptions;

export interface ColumnDropProps {
  gutter?: Gutter;
  minItemWidth?: MinItemWidth;
  noStretchedColumns?: boolean;
}

function getSafeMinItemWidth(
  theme: {
    space?: { [key: string]: string };
    sizes?: { [key: string]: string };
  },
  minItemWidth?: MinItemWidth,
) {
  return getSizeValue(theme, minItemWidth);
}

export const ColumnDrop = forwardRefWithAs<"div", ColumnDropProps>(
  (
    { as, gutter, style, minItemWidth, noStretchedColumns = false, ...props },
    ref,
  ) => {
    const theme = useTheme();
    const maybeGutter = getSafeGutter(theme, gutter);

    const attributeValue =
      noStretchedColumns === true ? "no-stretched-columns" : "";

    const safeMinItemWidth = getSafeMinItemWidth(theme, minItemWidth);
    const safeStyle = style ?? {};

    const Component = as ?? "div";
    return (
      <Component
        ref={ref}
        data-bedrock-column-drop={attributeValue}
        style={
          {
            ...safeStyle,
            "--gutter": maybeGutter,
            "--minItemWidth": safeMinItemWidth,
          } as CSSProperties
        }
        {...props}
      />
    );
  },
);
ColumnDrop.displayName = "ColumnDrop";
