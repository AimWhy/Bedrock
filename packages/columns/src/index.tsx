import { Gutter, getSafeGutter } from "@bedrock-layout/spacing-constants";
import { Stack, StackProps } from "@bedrock-layout/stack";
import {
  PolymorphicComponentPropsWithRef,
  PolymorphicRef,
} from "@bedrock-layout/type-utils";
import { useContainerQuery } from "@bedrock-layout/use-container-query";
import React, { ElementType, forwardRef } from "react";
import styled from "styled-components";

interface ColumnsBaseProps {
  gutter?: Gutter;
  columns?: number;
  dense?: boolean;
  forwardedAs?: unknown;
}

const ColumnsBase = styled.div.attrs<ColumnsBaseProps>(
  ({ dense, theme, gutter, columns = 1, style }) => {
    const maybeGutter = getSafeGutter(theme, gutter);
    const safeColumns = columns > 0 ? columns : 1;
    return {
      "data-bedrock-columns": dense ? "dense" : "",
      style: { ...style, "--gutter": maybeGutter, "--columns": safeColumns },
    };
  }
)<ColumnsBaseProps>`
  @property --gutter {
    syntax: "<length-percentage>";
    inherits: false;
    initial-value: 0;
  }

  @property --columns {
    syntax: "<number>";
    inherits: true;
    initial-value: 1;
  }

  box-sizing: border-box;
  > * {
    margin: 0;
  }

  display: grid;
  grid-template-columns: repeat(var(--columns, 1), 1fr);
  gap: var(--gutter, 0px);
  grid-auto-flow: row ${({ dense = false }) => (dense === true ? "dense" : "")};
`;

interface ColumnsPropsBase extends StackProps, ColumnsBaseProps {
  switchAt?: number | string;
  children?: React.ReactNode;
}

export type ColumnsProps<C extends ElementType = "div"> =
  PolymorphicComponentPropsWithRef<C, ColumnsPropsBase>;

const ColumnComp = forwardRef(
  <C extends ElementType = "div">(
    { columns, dense, switchAt, as, gutter, ...props }: ColumnsProps<C>,
    ref?: PolymorphicRef<C>
  ) => {
    const maybePx = React.useMemo(() => {
      return typeof switchAt === "string"
        ? toPX(switchAt)
        : typeof switchAt === "number" && switchAt > -1
        ? switchAt
        : null;
    }, [switchAt]);

    const widthToSwitchAt: number = maybePx ?? 0; //zero is used to make the switchAt a noop

    const [shouldSwitch, safeRef] = useContainerQuery(
      { width: widthToSwitchAt },
      ref
    );

    return shouldSwitch ? (
      <Stack as={as} ref={safeRef} gutter={gutter} {...props} />
    ) : (
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      //@ts-expect-error
      <ColumnsBase
        as={as}
        ref={safeRef}
        gutter={gutter}
        columns={columns}
        dense={dense}
        {...props}
      />
    );
  }
);

export const Columns = styled(ColumnComp).attrs(({ as, forwardedAs }) => {
  return {
    forwardedAs: as ?? forwardedAs,
    as: ColumnComp,
  };
})``;

Columns.displayName = "Columns";

export interface ColumnProps {
  span?: number;
  offsetStart?: number;
  offsetEnd?: number;
}

const safeSpan = (span: unknown) => {
  return typeof span === "number" ? span : 1;
};

/**
 *
 * span is remaped to colSpan due to span being an attribute that gets
 * passed to the underlying element.  This can cause issues with Grid layout.
 *
 * In a future breaking change, colSpan should be the public API.
 * */
export const Column = styled.div.attrs<ColumnProps>(
  ({ span, style, offsetStart = 0, offsetEnd = 0 }) => {
    const safeOffsetStart = offsetStart > 0 ? offsetStart : undefined;
    const safeOffsetEnd = offsetEnd > 0 ? offsetEnd : undefined;
    return {
      "data-bedrock-column": "",
      span: undefined,
      style: {
        ...style,
        "--span": Math.max(safeSpan(span), 1),
        "--offsetStart": safeOffsetStart,
        "--offsetEnd": safeOffsetEnd,
      },
    };
  }
)<ColumnProps>`
  @property --span {
    syntax: "<number>";
    inherits: true;
    initial-value: 1;
  }

  @property --offsetStart {
    syntax: "<number>";
    inherits: true;
    initial-value: 1;
  }

  @property --offsetEnd {
    syntax: "<number>";
    inherits: true;
    initial-value: 1;
  }

  grid-column: span min(var(--span, 1), var(--columns, 1));

  &[style*="--offset"] {
    display: contents;
  }

  &[style*="--offset"] > * {
    grid-column: span min(var(--span, 1), var(--columns, 1));
  }

  &[style*="--offsetStart"]::before {
    content: "";
    grid-column: span min(var(--offsetStart, 1), var(--columns, 1));
  }

  &[style*="--offsetEnd"]::after {
    content: "";
    grid-column: span min(var(--offsetEnd, 1), var(--columns, 1));
  }
`;

Column.displayName = "Column";

/**
 * This module is adapted from https://github.com/mikolalysenko/to-px/blob/master/browser.js
 */

const PIXELS_PER_INCH = 96;

/* c8 ignore start */
function parseUnit(str: string): [number, string] {
  str = String(str);
  const num = parseFloat(str);

  /* c8 ignore next */
  const [, unit] = str.match(/[\d.\-+]*\s*(.*)/) ?? ["", ""];

  return [num, unit];
}

/* c8 ignore next */
function toPX(str: string, element?: Element): number | null {
  /* c8 ignore next */
  if (!str) return null;

  const elementOrBody = element ?? document.body;
  const safeStr = (str ?? "px").trim().toLowerCase();

  /* c8 ignore next */
  switch (safeStr) {
    case "vmin":
    case "vmax":
    case "vh":
    case "vw":
    case "%":
      return null;
    case "ch":
    case "ex":
      return getSizeBrutal(safeStr, elementOrBody);
    case "em":
      return getPropertyInPX(elementOrBody, "font-size");
    case "rem":
      return getPropertyInPX(document.body, "font-size");
    case "in":
      return PIXELS_PER_INCH;
    case "cm":
      return PIXELS_PER_INCH / 2.54;
    case "mm":
      return PIXELS_PER_INCH / 25.4;
    case "pt":
      return PIXELS_PER_INCH / 72;
    case "pc":
      return PIXELS_PER_INCH / 6;
    case "px":
      return 1;
    default: {
      const [value, units] = parseUnit(safeStr);

      if (isNaN(value)) return null;

      if (!units) return value;

      const px = toPX(units, element);
      return typeof px === "number" ? value * px : null;
    }
  }
}

/* c8 ignore next */
function getPropertyInPX(element: Element, prop: string): number {
  const [value, units] = parseUnit(
    getComputedStyle(element).getPropertyValue(prop)
  );
  return value * (toPX(units, element) ?? 1);
}

function getSizeBrutal(unit: string, element: Element) {
  const testDIV = document.createElement("div");
  testDIV.style["height"] = "128" + unit;
  element.appendChild(testDIV);
  const size = getPropertyInPX(testDIV, "height") / 128;
  element.removeChild(testDIV);
  return size;
}
/* c8 ignore stop */
