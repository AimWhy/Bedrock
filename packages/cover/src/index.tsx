import {
  CSSLength,
  Gutter,
  SizesOptions,
  getSafeGutter,
  getSizeValue,
  useTheme,
} from "@bedrock-layout/spacing-constants";
import { forwardRefWithAs } from "@bedrock-layout/type-utils";
import React, { CSSProperties } from "react";

/**
 * `minHeight` can be a CSSLength, a number, or a key of the theme's sizes object
 */
export type MinHeight = CSSLength | number | SizesOptions;

/**
 * Props for the Cover component.
 */
export type CoverProps = {
  /**
   * Slot to be placed at the top of the cover component, above the centered content.
   */
  top?: React.ReactNode;
  /**
   * Slot to be placed at the bottom of the cover component, below the centered content.
   */
  bottom?: React.ReactNode;
  /**
   * Sets space between each element.
   * @deprecated Use `gap` instead.
   */
  gutter?: Gutter;
  /**
   * Sets space between each element.
   */
  gap?: Gutter;
  /**
   * Sets the minimum height of the cover component.
   * `minHeight` can be a CSSLength, a number, or a key of the theme's sizes object
   * @default "100vh"
   */
  minHeight?: MinHeight;
  /**
   * Sets the content to stretch to the full height of the cover component minus the top and bottom slots.
   */
  stretchContent?: boolean;
};

/**
 * The `Cover` component is designed to vertically cover a predefined area, `100vh` by default, and vertically center its children.
 * You can also conditionally render a top and/or bottom slot as well.
 */
export const Cover = forwardRefWithAs<"div", CoverProps>(function Cover(
  {
    as: Component = "div",
    children,
    gap,
    gutter,
    top,
    bottom,
    minHeight,
    style = {},
    //TODO: Change from boolean to string union of "stretch-content" | "default"
    stretchContent,
    ...props
  },
  ref,
) {
  const theme = useTheme();
  const maybeGutter = getSafeGutter(theme, gap ?? gutter);
  const maybeMinHeight = getSizeValue(theme, minHeight);

  const attributeVal = stretchContent === true ? "stretch-content" : "";

  return (
    <Component
      ref={ref}
      data-bedrock-cover={attributeVal}
      style={
        {
          "--gutter": maybeGutter,
          "--minHeight": maybeMinHeight,
          ...style,
        } as CSSProperties
      }
      {...props}
    >
      {top && <div data-bedrock-cover-top="">{top}</div>}
      <div data-bedrock-cover-centered="">{children}</div>
      {bottom && <div data-bedrock-cover-bottom="">{bottom}</div>}
    </Component>
  );
});
