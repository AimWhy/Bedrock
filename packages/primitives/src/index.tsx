export * from "@bedrock-layout/appboundary";
export * from "@bedrock-layout/center";
export * from "@bedrock-layout/column-drop";
export * from "@bedrock-layout/columns";
export * from "@bedrock-layout/cover";
export * from "@bedrock-layout/frame";
export * from "@bedrock-layout/grid";
export * from "@bedrock-layout/inline";
export * from "@bedrock-layout/inline-cluster";
export * from "@bedrock-layout/masonry-grid";
export * from "@bedrock-layout/padbox";
export * from "@bedrock-layout/reel";
export * from "@bedrock-layout/register-resize-callback";
export * from "@bedrock-layout/spacing-constants";
export * from "@bedrock-layout/split";
export * from "@bedrock-layout/stack";
export { useContainerQuery } from "@bedrock-layout/use-container-query";
export { useForwardedRef } from "@bedrock-layout/use-forwarded-ref";
export { useResizeObserver } from "@bedrock-layout/use-resize-observer";
export { useStatefulRef } from "@bedrock-layout/use-stateful-ref";
