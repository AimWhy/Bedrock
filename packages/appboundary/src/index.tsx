import { Center } from "@bedrock-layout/center";
import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
import React from "react";
import styled, { ThemeContext } from "styled-components";

const safeTheme = { breakPoints: {} };

export const AppBoundary = styled.div.attrs((props) => {
  const { breakPoints } = React.useContext(ThemeContext) ?? safeTheme;
  const maxWidth = mergeBreakpoints(breakPoints).xxlarge;
  return {
    "data-bedrock-layout-appboundary": "",
    children: <Center maxWidth={maxWidth}>{props.children}</Center>,
  };
})`
  padding: 0;
  max-width: 100%;
  overflow: hidden;
  height: 100%;
`;

AppBoundary.displayName = "AppBoundary";
