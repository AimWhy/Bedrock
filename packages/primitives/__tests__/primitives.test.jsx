import * as primitives from "../src";

describe("primitives", () => {
  test("primitives snapshot", () => {
    expect(primitives).toMatchSnapshot();
  });
});
