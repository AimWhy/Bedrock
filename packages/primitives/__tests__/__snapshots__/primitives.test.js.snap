// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primitives primitives snapshot 1`] = `
Object {
  "AppBoundary": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -1381688698,
      "baseStyle": undefined,
      "componentId": "sc-gsDKAQ",
      "isStatic": false,
      "rules": Array [
        "
  padding: 0;
  max-width: 100%;
  overflow: hidden;
  height: 100%;
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "boundarySize": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-gsDKAQ",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Center": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -578376589,
      "baseStyle": undefined,
      "componentId": "sc-bdvvtL",
      "isStatic": false,
      "rules": Array [
        "
  @property --maxWidth {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: 100%;
  }

  box-sizing: content-box;

  margin-inline-start: auto;
  margin-inline-end: auto;
  margin-inline: auto;

  max-inline-size: var(--maxWidth, 100%);

  ",
        [Function],
        "

  ",
        [Function],
        "
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "centerChildren": [Function],
      "centerText": [Function],
      "maxWidth": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-bdvvtL",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Column": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": 605235812,
      "baseStyle": undefined,
      "componentId": "sc-gKclnd",
      "isStatic": false,
      "rules": Array [
        "
  --span: ",
        [Function],
        ";

  ",
        [Function],
        "

  ",
        [Function],
        "

  ",
        [Function],
        "
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "span": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-gKclnd",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "ColumnDrop": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -1179868038,
      "baseStyle": undefined,
      "componentId": "sc-dkPtRN",
      "isStatic": false,
      "rules": Array [
        "
  @property --basis {
    syntax: \\"<length-percentage>\\";
    inherits: true;
    initial-value: ",
        "159px",
        ";
  }

  @property --gutter {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: 0px;
  }

  box-sizing: border-box;
  > * {
    margin: 0;
    flex-basis: var(--basis, ",
        "159px",
        ");
    flex-grow: ",
        [Function],
        ";
    flex-shrink: 1;
  }

  display: flex;
  flex-wrap: wrap;
  gap: var(--gutter, 0px);
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "basis": [Function],
      "gutter": [Function],
      "noStretchedColumns": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-dkPtRN",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Columns": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -1887681386,
      "baseStyle": undefined,
      "componentId": "sc-jRQBWg",
      "isStatic": false,
      "rules": Array [
        "",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "columns": [Function],
      "dense": [Function],
      "gutter": [Function],
      "switchAt": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-jRQBWg",
    "target": Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    },
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Cover": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": 1551091733,
      "baseStyle": undefined,
      "componentId": "sc-iCfMLu",
      "isStatic": false,
      "rules": Array [
        "
  @property --gutter {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: 0;
  }

  @property --minHeight {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: 100vh;
  }

  > * {
    margin: 0;
  }

  display: flex;
  flex-direction: column;
  gap: var(--gutter, 0px);

  min-block-size: var(--minHeight, 100vh);

  > [data-bedrock-cover-centered] {
    margin-block-start: auto;
    margin-block-end: auto;

    ",
        [Function],
        ";
  }
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "bottom": [Function],
      "gutter": [Function],
      "minHeight": [Function],
      "stretchContent": [Function],
      "top": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-iCfMLu",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Frame": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": 1337550442,
      "baseStyle": undefined,
      "componentId": "sc-furwcr",
      "isStatic": false,
      "rules": Array [
        "
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;

  &[style*=\\"--ratio\\"] {
    aspect-ratio: var(--ratio);
  }

  > * {
    position: absolute;

    inset-block-start: 0;
    inset-block-end: 0;
    inset-inline-start: 0;
    inset-inline-end: 0;

    inset-block: 0;
    inset-inline: 0;

    display: flex;
    justify-content: center;
    align-items: center;
  }

  > :is(img, video) {
    inline-size: 100%;
    block-size: 100%;
    size: 100%;

    object-fit: cover;
    object-position: ",
        [Function],
        ";
  }
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "position": [Function],
      "ratio": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-furwcr",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "GlobalStyles": Object {
    "$$typeof": Symbol(react.memo),
    "compare": null,
    "type": [Function],
  },
  "Grid": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": 1136856117,
      "baseStyle": undefined,
      "componentId": "sc-pVTFL",
      "isStatic": false,
      "rules": Array [
        "
  @property --gutter {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: 0;
  }

  @property --minItemWidth {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: ",
        "639px",
        ";
  }

  box-sizing: border-box;
  > * {
    margin: 0;
  }

  display: grid;
  gap: var(--gutter, 0px);

  grid-template-columns: repeat(
    auto-fit,
    minmax(min(var(--minItemWidth, ",
        "639px",
        "), 100%), 1fr)
  );
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "gutter": [Function],
      "minItemWidth": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-pVTFL",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Inline": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
      [Function],
    ],
    "componentStyle": e {
      "baseHash": 851465943,
      "baseStyle": e {
        "baseHash": -620189328,
        "baseStyle": undefined,
        "componentId": "sc-jrQzAO",
        "isStatic": false,
        "rules": Array [
          "
  --gutter: ",
          [Function],
          ";

  box-sizing: border-box;
  > * {
    margin: 0;
  }

  display: flex;
  flex-wrap: wrap;
  gap: var(--gutter);

  justify-content: ",
          [Function],
          ";

  align-items: ",
          [Function],
          ";
",
        ],
        "staticRulesId": "",
      },
      "componentId": "sc-kDTinF",
      "isStatic": false,
      "rules": Array [
        "
  flex-wrap: nowrap;
  ",
        [Function],
        "
  ",
        [Function],
        "
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [
      "sc-jrQzAO",
    ],
    "propTypes": Object {
      "align": [Function],
      "gutter": [Function],
      "justify": [Function],
      "stretch": [Function],
      "switchAt": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-kDTinF",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "InlineCluster": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -620189328,
      "baseStyle": undefined,
      "componentId": "sc-jrQzAO",
      "isStatic": false,
      "rules": Array [
        "
  --gutter: ",
        [Function],
        ";

  box-sizing: border-box;
  > * {
    margin: 0;
  }

  display: flex;
  flex-wrap: wrap;
  gap: var(--gutter);

  justify-content: ",
        [Function],
        ";

  align-items: ",
        [Function],
        ";
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "align": [Function],
      "gutter": [Function],
      "justify": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-jrQzAO",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "MasonryGrid": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -658045346,
      "baseStyle": e {
        "baseHash": 1136856117,
        "baseStyle": undefined,
        "componentId": "sc-pVTFL",
        "isStatic": false,
        "rules": Array [
          "
  @property --gutter {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: 0;
  }

  @property --minItemWidth {
    syntax: \\"<length-percentage>\\";
    inherits: false;
    initial-value: ",
          "639px",
          ";
  }

  box-sizing: border-box;
  > * {
    margin: 0;
  }

  display: grid;
  gap: var(--gutter, 0px);

  grid-template-columns: repeat(
    auto-fit,
    minmax(min(var(--minItemWidth, ",
          "639px",
          "), 100%), 1fr)
  );
",
        ],
        "staticRulesId": "",
      },
      "componentId": "sc-crHmcD",
      "isStatic": false,
      "rules": Array [
        "
  grid-template-rows: 1px;
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [
      "sc-pVTFL",
    ],
    "propTypes": Object {
      "gutter": [Function],
      "minItemWidth": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-crHmcD",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "PadBox": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -225017612,
      "baseStyle": undefined,
      "componentId": "sc-egiyK",
      "isStatic": false,
      "rules": Array [
        "
  box-sizing: border-box;
  ",
        [Function],
        "
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "padding": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-egiyK",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Split": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -1947125077,
      "baseStyle": undefined,
      "componentId": "sc-ksdxgE",
      "isStatic": false,
      "rules": Array [
        "",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "fraction": [Function],
      "gutter": [Function],
      "switchAt": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-ksdxgE",
    "target": Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    },
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Stack": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [
      [Function],
    ],
    "componentStyle": e {
      "baseHash": -1280029055,
      "baseStyle": undefined,
      "componentId": "sc-hKwDye",
      "isStatic": false,
      "rules": Array [
        "
  box-sizing: border-box;
  > * {
    margin: 0;
  }
  --gutter: ",
        [Function],
        ";

  display: grid;
  gap: var(--gutter);
  align-content: start;

  & > [data-bedrock-column] {
    grid-column: span 1 / auto;
  }
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": Array [],
    "propTypes": Object {
      "gutter": [Function],
    },
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-hKwDye",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "breakPoints": Object {
    "large": 1199,
    "largeOnly": Array [
      1024,
      1199,
    ],
    "medium": 1023,
    "mediumOnly": Array [
      640,
      1023,
    ],
    "smallOnly": 639,
    "xlarge": 1439,
    "xlargeOnly": Array [
      1200,
      1439,
    ],
    "xxlarge": 1920,
  },
  "checkIsCSSLength": [Function],
  "getSizeValue": [Function],
  "getSpacingValue": [Function],
  "mergeBreakpoints": [Function],
  "reset": Array [
    "
  /*
  reset.css
*/

  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }

  * {
    margin: 0;
  }

  ul[class],
  ol[class] {
    padding: 0;
    list-style: none;
  }

  html:focus-within {
    scroll-behavior: smooth;
  }

  body {
    min-block-size: 100vh;
    text-rendering: optimizeSpeed;
    line-height: 1.5;
    font-size: 100%;
    -webkit-font-smoothing: antialiased;
  }

  a:not([class]) {
    text-decoration-skip-ink: auto;
  }

  img,
  picture {
    display: block;
    max-inline-size: 100%;
  }

  p,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  li,
  figcaption {
    max-inline-size: 60ch;
    overflow-wrap: break-word;
  }

  input,
  button,
  textarea,
  select {
    font: inherit;
    max-inline-size: 100%;
  }

  .js-focus-visible :focus:not(.focus-visible) {
    outline: none;
  }

  @media (prefers-reduced-motion: reduce) {
    html:focus-within {
      scroll-behavior: auto;
    }

    *,
    *::before,
    *::after {
      animation-duration: 0.01ms !important;
      animation-iteration-count: 1 !important;
      transition-duration: 0.01ms !important;
      scroll-behavior: auto !important;
    }
  }
",
  ],
  "sizes": Object {
    "large": "1199px",
    "medium": "1023px",
    "small": "639px",
    "xlarge": "1439px",
    "xsmall": "319px",
    "xxlarge": "1920px",
    "xxsmall": "159px",
  },
  "spacing": Object {
    "lg": "1rem",
    "lgXl": "1.5rem",
    "md": "0.5rem",
    "mdLg": "0.75rem",
    "none": "0px",
    "sm": "0.25rem",
    "xl": "2rem",
    "xlXXl": "3rem",
    "xs": "0.125rem",
    "xxl": "4rem",
    "xxs": "0.0625rem",
  },
  "toPX": [Function],
  "useContainerQuery": [Function],
  "useForwardedRef": [Function],
  "useResizeObserver": [Function],
  "useStatefulRef": [Function],
}
`;
